update-script.js

```diff
@@ -0,0 +1,141 @@
+
+/**
+ * Fetches the source records from the APPLE_API_ENDPOINT
+ *
+ * @return {String[][]} 
+ */
+const getSourceRecords = async () => {
+  const response = await fetch(APPLE_API_ENDPOINT, {
+    headers: {
+      "Accept": "application/vnd.github.v3.raw"
+    }
+  });
+  const data = await response.json();
+  return data;
+}
+
+const arrayEquals = (a, b) => {
+  return Array.isArray(a) &&
+    Array.isArray(b) &&
+    a.length === b.length &&
+    a.every((val, index) => val === b[index]);
+};
+
+/**
+ * Updates the existing record in Remote Settings with the updated data from Apple's GitHub repository
+ *
+ * @param {KintoClient} client KintoClient instance
+ * @param {string} bucket Name of the Remote Settings bucket
+ * @param {Object} newRecord Object containing the updated related realms object
+ * @param {string} newRecord.id ID from the current related realms object from the Remote Settings server
+ * @param {string[][]} newRecord.relatedRealms Updated related realms array from GitHub
+ */
+const updateRecord = async (client, bucket, newRecord) => {
+  await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).updateRecord(newRecord);
+  const postServerData = await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).getData();
+  const setDataObject = {
+    status: "to-review",
+    last_modified: postServerData.last_modified
+  };
+  await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).setData(setDataObject, {patch: true});
+  console.log(`Found new records, committing changes to ${WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION} collection.`);
+};
+
+/**
+ * Creates a new record in Remote Settings if there are no records in the WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION
+ *
+ * @param {KintoClient} client
+ * @param {string} bucket
+ */
+const createRecord = async (client, bucket) => {
+  let githubRecords = await getSourceRecords();
+  const result = await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).createRecord({
+    relatedRealms: githubRecords
+  });
+  const postServerData = await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).getData();
+  await client.bucket(bucket).collection(WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION).setData({status: "to-review", last_modified: postServerData.last_modified}, {patch: true});
+  console.log(`Added new record to ${WEBSITES_WITH_SHARED_CREDENTIAL_COLLECTION}`, result);
+};
+
+const printSuccessMessage = () => {
+  console.log("Script finished successfully!");
+}
+
+/**
+ * The runner for the script.
+ * 
+ * @return {Number} 0 for success, 1 for failure.
+ */
+const main = async () => {
+  if (FX_RS_WRITER_USER === "" || FX_RS_WRITER_PASS === "") {
+    console.error("No username or password set, quitting!");
+    return 1;
+  }
+  const bucket = "main-workspace";
```

Comment:
this could be a constant

Reply:
Moved this to global scope. Should it be an `AppConstant` instead?

Reply:
Same as the collection id I think